### PR TITLE
add docstrings for 2 methods in Grid

### PR DESCRIPTION
**Pull request by gracetangg**
I added docstrings to the Grid methods, but I'm not sure if I got the right idea about how it works.

Address #28

### CODE REVIEW

File: turbopy/core.py [hunk 1]
```diff
@@ -337,6 +337,14 @@ def is_running(self):
 
 
 class Grid:
+    """Grid class
```
Comment (padamson) [inline turbopy/core.py:340]: I added placeholder docstrings for Grid and grid_data so your additions would show up when the documentation is built. If there is no class level docstring, Sphinx will not use the lower level docstrings.  Also, I added a placeholder for the grid_data attribute to demonstrate cross referencing the object in lower level documentation.

File: turbopy/core.py [hunk 2]
```diff
@@ -345,7 +353,8 @@ def __init__(self, grid_data: dict):
         self.dr = None
         self.parse_grid_data()
 
-        self.r = self.r_min + (self.r_max - self.r_min) * self.generate_linear()
+        self.r = (self.r_min + (self.r_max - self.r_min) *
```
Comment (padamson) [inline turbopy/core.py:356]: My linter flagged this line as being too long. It's ok to clean up code when adding docs.

File: turbopy/core.py [hunk 3]
```diff
@@ -354,6 +363,16 @@ def __init__(self, grid_data: dict):
         self.r_inv[0] = 0
 
     def parse_grid_data(self):
+        """
+        Initializes the grid spacing, range, and number of points on the grid
+        from :class:`Grid.grid_data`.
```
Comment (padamson) [inline turbopy/core.py:368]: Notice the cross reference to the class attribute.

File: turbopy/core.py [hunk 4]
```diff
@@ -363,10 +382,29 @@ def parse_grid_data(self):
...
+        ----------
+        var_name : str
+            Attribute name to be initialized.
+        options : set
+            Set of keys in :class:`Grid.grid_data` to search for values.
+
+        Raises
+        ------
+        KeyError
+            If none of the keys in `options` are present in
```
Comment (padamson) [inline turbopy/core.py:405]: I modified some of your documentation slightly.

**Comment by padamson** [comment]
@gracetangg AWESOME job on your first contribution to turbopy docs.  Please look at my commit and make sure you agree/understand my changes to your submission. I'm happy to discuss with you. Since I co-authored with you, I'm asking @arichar6 for a review.

File: turbopy/core.py (hunk 1 above)
Comment (arichar6) [reply turbopy/core.py:340]: Thanks! This made me confused for a while on a different project, until I realized that Sphinx needed the class level docstring.

File: turbopy/core.py [hunk 5]
```diff
@@ -363,10 +382,29 @@ def parse_grid_data(self):
             self.set_value_from_keys("dr", {"dr", "dx"})
             self.num_points = 1 + (self.r_max - self.r_min) / self.dr
             if not (self.num_points % 1 == 0):
-                raise (RuntimeError("Invalid grid spacing: configuration does not imply integer number of grid points"))
+                raise (RuntimeError("Invalid grid spacing: configuration "
```
Comment (arichar6) [inline turbopy/core.py:385]: Wow! I had no idea you could join strings this way! TIL new syntax for defining long strings.

**Pull request merged**